src/applications/MachineShopSimulator.java

```diff
@@ -29,7 +29,101 @@ public static void main(String[] args) {
         MachineShopSimulator simulator = new MachineShopSimulator();
-        SimulationResults simulationResults = specification.runSimulation(simulator);
+        SimulationResults simulationResults = simulator.runSimulation(specification);
         simulationResults.print();
     }
+
+	void simulate(SimulationResults simulationResults) {
+	    while (numJobs > 0) {// at least one job left
+	        int nextToFinish = eList.nextEventMachine();
+	        timeNow = eList.nextEventTime(nextToFinish);
+			// schedule next one.
+            Job lastJob;
+            Machine nextMachine = machine[nextToFinish];
+			if (nextMachine.getActiveJob() == null) {// in idle or change-over
+			                                            // state
+			    lastJob = null;
+			    // wait over, ready for new job
+			    if (nextMachine.jobQisEmpty()) // no waiting job
+			        eList.setFinishTime(nextToFinish, largeTime);
+			    else {// take job off the queue and work on it
+			        nextMachine.setActiveJob((Job) nextMachine.getJobQ()
+			                .remove());
+			        nextMachine.setTotalWait(nextMachine.getTotalWait() + timeNow
+			                - nextMachine.getActiveJob().getArrivalTime());
+			        nextMachine.setNumTasks(nextMachine.getNumTasks() + 1);
+			        int t = nextMachine.getActiveJob().removeNextTask();
+			        eList.setFinishTime(nextToFinish, timeNow + t);
+			    }
+			} else {// task has just finished on machine[theMachine]
+			        // schedule change-over time
+			    lastJob = nextMachine.getActiveJob();
+			    nextMachine.setActiveJob(null);
+			    eList.setFinishTime(nextToFinish, timeNow
+			            + nextMachine.getChangeTime());
+			}
+	        // change job on machine nextToFinish
+	        Job theJob = lastJob;
+	        // move theJob to its next machine
+	        // decrement numJobs if theJob has finished
+	        if (theJob != null && !theJob.moveToNextMachine(this, simulationResults))
+	            numJobs--;
+	    }
+    }
+    public  SimulationResults runSimulation(SimulationSpecification simulationSpecification) {
+	    largeTime = Integer.MAX_VALUE;
+	    timeNow = 0;
+	    simulationSpecification.startShop(this); // initial machine loading
+	    SimulationResults simulationResults = new SimulationResults(getNumJobs());
+	    simulate(simulationResults); // run all jobs through shop
+	    simulationResults.outputStatistics(this);
+	    return simulationResults;
+	}
+    // getters and setters
+    public int getTimeNow(){
+        return timeNow;
+    }
+  
+    public int getNumMachines() {
+        return numMachines;
+    }
+
+    public void setNumMachines(int numMachines) {
+        this.numMachines = numMachines;
+    }
+
+    public int getNumJobs() {
+        return numJobs;
+    }
+
+    public void setNumJobs(int numJobs) {
+        this.numJobs = numJobs;
+    }
+
+    public EventList geteList() {

```

Comment:
I feel like this function's name isn't great. When I first saw it somewhere I though "what is a _gete_ list?"
I think getEList and getEventList would both be better names.

Reply:
Yeah I noticed that too